### PR TITLE
generate parse_data and parse_file_stream

### PR DESCRIPTION
**Pull request by tanyifeng**
Generate function of parse_data and parse_file_stream, so we can parse json data of char* or file stream.

Signed-off-by: Yifeng Tan <tanyifeng1@huawei.com>

### CODE REVIEW

File: src/generate.py [hunk 1]
```diff
@@ -593,6 +595,61 @@ def generate_C_epilogue(c_file, prefix):
...
+        return NULL;
+    }
+    tree = yajl_tree_parse (content, errbuf, sizeof(errbuf));
+    free (content);
+    if (tree == NULL) {
+        *err = strdup ("cannot parse the file");
+        return NULL;
+    }
+
+    oci_%s_%s *container = make_oci_%s_%s (tree, ctx, err);
```
Comment (giuseppe) [inline src/generate.py]: since we made it more generic, should we use something else than `container`?

File: src/generate.py [hunk 2]
```diff
@@ -593,6 +595,61 @@ def generate_C_epilogue(c_file, prefix):
...
+        *err = strdup ("oci data can not be NULL");
+        return NULL;
+    }
+    tree = yajl_tree_parse (jsondata, errbuf, sizeof(errbuf));
+    if (tree == NULL) {
+        *err = strdup ("cannot parse the oci data");
+        return NULL;
+    }
+
+    oci_%s_%s *container = make_oci_%s_%s (tree, ctx, err);
```
Comment (giuseppe) [inline src/generate.py]: same here

**Comment by giuseppe** [comment]
could you please rebase this PR?  I've added Travis integration so we will be able to run tests as part of the PR as well :-)

**Comment by tanyifeng** [comment]
fixed

**Pull request merged**